pkg/executor/plugins/grpc/server.go

```diff
@@ -0,0 +1,110 @@
+package grpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/bacalhau-project/bacalhau/pkg/bidstrategy"
+	"github.com/bacalhau-project/bacalhau/pkg/executor"
+	"github.com/bacalhau-project/bacalhau/pkg/executor/plugins/grpc/proto"
+	"github.com/bacalhau-project/bacalhau/pkg/model"
+)
+
+type GRPCServer struct {
+	Impl executor.Executor
+}
+
+func (s *GRPCServer) Run(ctx context.Context, request *proto.RunCommandRequest) (*proto.RunCommandResponse, error) {
+	args := new(executor.RunCommandRequest)
+	if err := json.Unmarshal(request.Params, args); err != nil {
+		return nil, err
+	}
+	result, err := s.Impl.Run(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+	b, err := json.Marshal(result)
+	if err != nil {
+		return nil, err
+	}
+	return &proto.RunCommandResponse{Params: b}, nil
```

Comment:
this is a blocking call and the rpc request will return after the execution is completed, which can be an hour today, and days in the future. Wouldn't that cause issues? I understand this is the case today with our current executors, but those are in-process executors and are running on a separate goroutine

Reply:
Added TODO as code comment linking to issue: https://github.com/bacalhau-project/bacalhau/issues/2702